src/FontEmoticonInfo.php

```diff
@@ -19,37 +19,84 @@ class FontEmoticonInfo
      */
     private $m_htmlCode;
 
+    /**
+     * The name of the smiley.
+     * @var string
+     */
+    private $m_name;
+
+    /**
+     * A concatenation of regex-escaped smiley representations.
+     * @var string
+     */
+    private $m_smileyRegex;
+
     public function __construct($name, $textRepresentations)
     {
-        $this->m_regex = '';
-        $is_first = true;
+        $this->m_name = $name;
+
+        // Ensure that each ASCII representation is quoted separately.
+        $this->m_smileyRegex = join( '|', array_map( function( $smiley ) {
+            return preg_quote( $smiley, '/' );
+        } ) );
```

Comment:
There is a parameter ($textRepresentations) missing here.

Reply:
Thanks. Fixed in the new push.